### PR TITLE
A minidocs transform

### LINKED ISSUE #42: Create a minidocs transform
A minidocs transform could look like this:

- it could live in the minidocs repo at `/transform.js`
- basic usage of minidocs would be easier. users wouldn't have to install and use read-directory or myriad other transforms, they could just use minidocs/transform: 
  - `browserify index.js -t minidocs/transform -o bundle.js`
- it would help a lot with optimizing the build to get a smaller bundle size without making the users figure that out
- it could handle:
  - reading, parsing, and highlighting markdown with `marked` & `highlight.js`
  - parsing the table of contents into the format the sidebar menu uses
  - inlining/bundling the css via `sheetify`
  - running `yo-yoify` for older browser support

I think @mathisonian suggested a minidocs-specific transform a while ago 😄. This morning I was like "oh right that's a good idea."

I'm looking at bundleify as inspiration for how this might work: https://github.com/bendrucker/bundleify

### PR DESCRIPTION
**Pull request by sethvincent**
This adds a transform for minidocs, which simplifies usage and helps to bring down the bundle size. 

This closes #42 and may address #39, though there may still be things that can be done to reduce bundle size.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -51,19 +51,19 @@ The folder `site` will now contain the `html` `js` and `css` for your site.
 
 ### library
 
-Specify a table of contents
+Create a table of contents in a file named `contents.json`:
 
 ```javascript
```
Comment (yoshuawuyts) [inline README.md]: change to `json`? believe it shows up slightly differently in the highlighting

Comment (sethvincent) [inline README.md]: woops, thanks!

**Comment by freeman-lab** [comment]
did some local testing with this and all seems to be working

**Comment by freeman-lab** [comment]
woot woot!

**Pull request merged**